onset_maproom/layout.py

```diff
@@ -22,19 +29,67 @@
 LIGHT_GRAY = "#eeeeee"
 
 
+def adm_borders(shapes):
+    with psycopg2.connect(**CONFIG["db"]) as conn:
+        s = sql.Composed(
+            [
+                sql.SQL("with g as ("),
+                sql.SQL(shapes),
+                sql.SQL(
+                    """
+                    )
+                    select
+                        g.label, g.key, g.the_geom
+                    from g
+                    """
+                ),
+            ]
+        )
+        df = pd.read_sql(s, conn)
+
+    df["the_geom"] = df["the_geom"].apply(lambda x: wkb.loads(x.tobytes()))
+    df["the_geom"] = df["the_geom"].apply(
+        lambda x: x if isinstance(x, MultiPolygon) else MultiPolygon([x])
+    )
+    shapes = df["the_geom"].apply(shapely.geometry.mapping)
+    for i in df.index: #this adds the district layer as a label in the dict
+        shapes[i]['label'] = df['label'][i]
+    return {"features": shapes}
+
+
+def make_adm_overlay(adm_list, adm_lev, adm_weight):
+    border_id = f'borders_adm{adm_lev}'
+    adm_check = False
+    if adm_lev == 1:
+        adm_check = True
+    return dlf.Overlay(
+        dlf.GeoJSON(
+            id=border_id,
+            data=adm_borders(adm_list[adm_lev-1]["sql"]),
+            options={
+                "fill": False,
+                "color": adm_list[adm_lev-1]["color"],
+                "weight": adm_weight,
+            },
+        ),
+        name=adm_list[adm_lev-1]["name"],
+        checked=adm_check,
+    )
+
+
 def app_layout():
 
     # Initialization
     rr_mrg = calc.read_zarr_data(RR_MRG_ZARR)
     center_of_the_map = [((rr_mrg.Y[0]+rr_mrg.Y[-1])/2).values, ((rr_mrg.X[0]+rr_mrg.X[-1])/2).values]
     lat_res = np.around((rr_mrg.Y[1]-rr_mrg.Y[0]).values, decimals=10)
-    lat_min = str(np.around((rr_mrg.Y[0]-lat_res/2).values, decimals=10))
-    lat_max = str(np.around((rr_mrg.Y[-1]+lat_res/2).values, decimals=10))
+    lat_min = np.around((rr_mrg.Y[0]-lat_res/2).values, decimals=10)
```

Comment:
It seems to be needed to avoid printing machine error values. See screenshot of an example. 10 is probably arbitrary. Might be needed for very high resolution stuff... That's probably why I went so far. Maybe there is a better way to go around the machine error problem?

![Screen Shot 2022-09-29 at 11 54 06 AM](https://user-images.githubusercontent.com/42619415/193080156-3229be25-993e-4ce7-a391-7e27cbb12e71.png)